src/goto-symex/builtin_functions.cpp

```diff
@@ -467,10 +467,7 @@ void goto_symext::intrinsic_switch_to(
   // Switch to other thread.
   const expr2tc &num = call.operands[0];
   if(!is_constant_int2t(num))
-  {
-    std::cerr << "Can't switch to non-constant thread id no";
-    abort();
-  }
+    throw std::runtime_error("Can't switch to non-constant thread id no");
```

Comment:
`Can't switch to non-constant thread id no` -> `Can't switch to non-constant thread id number`. Shall we output the number?

Reply:
We don't know if it is a number. I've changed to output the contents of expr as well